zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,978 @@
+    use rand::{RngCore, SeedableRng};
+    use rand_xorshift::XorShiftRng;
+
+    #[derive(Clone, Debug)]
+    struct TestCircuitSignVerifyConfig<F: FieldExt> {
+        sign_verify: SignVerifyConfig<F>,
+    }
+
+    impl<F: FieldExt> TestCircuitSignVerifyConfig<F> {
+        pub fn new(meta: &mut ConstraintSystem<F>) -> Self {
+            let power_of_randomness = {
+                let columns = [(); POW_RAND_SIZE].map(|_| meta.instance_column());
+                let mut power_of_randomness = None;
+
+                meta.create_gate("power of randomness", |meta| {
+                    power_of_randomness =
+                        Some(columns.map(|column| meta.query_instance(column, Rotation::cur())));
+
+                    [0.expr()]
+                });
+
+                power_of_randomness.unwrap()
+            };
+
+            let sign_verify = SignVerifyConfig::new(meta, power_of_randomness);
+            TestCircuitSignVerifyConfig { sign_verify }
+        }
+    }
+
+    #[derive(Default)]
+    struct TestCircuitSignVerify<F: FieldExt, const MAX_VERIF: usize> {
+        sign_verify: SignVerifyChip<F, MAX_VERIF>,
+        randomness: F,
+        txs: Vec<SignData>,
+    }
+
+    impl<F: FieldExt, const MAX_VERIF: usize> Circuit<F> for TestCircuitSignVerify<F, MAX_VERIF> {
+        type Config = TestCircuitSignVerifyConfig<F>;
+        type FloorPlanner = SimpleFloorPlanner;
+
+        fn without_witnesses(&self) -> Self {
+            Self::default()
+        }
+
+        fn configure(meta: &mut ConstraintSystem<F>) -> Self::Config {
+            TestCircuitSignVerifyConfig::new(meta)
+        }
+
+        fn synthesize(
+            &self,
+            config: Self::Config,
+            mut layouter: impl Layouter<F>,
+        ) -> Result<(), Error> {
+            self.sign_verify.assign_txs(
+                &config.sign_verify,
+                &mut layouter,
+                self.randomness,
+                &self.txs,
+            )?;
+            Ok(())
+        }
+    }
+
+    fn run<F: FieldExt, const MAX_VERIF: usize>(k: u32, txs: Vec<SignData>) {
+        let mut rng = XorShiftRng::seed_from_u64(2);
+        let aux_generator =
+            <Secp256k1Affine as CurveAffine>::CurveExt::random(&mut rng).to_affine();
+
+        let randomness = F::random(&mut rng);
+        let mut power_of_randomness: Vec<Vec<F>> = (1..POW_RAND_SIZE + 1)
+            .map(|exp| vec![randomness.pow(&[exp as u64, 0, 0, 0]); txs.len() * VERIF_HEIGHT])
+            .collect();
+        // SignVerifyChip -> ECDSAChip -> MainGate instance column
+        power_of_randomness.push(vec![]);
+        let circuit = TestCircuitSignVerify::<F, MAX_VERIF> {
+            sign_verify: SignVerifyChip {
+                aux_generator,
+                window_size: 2,
+                _marker: PhantomData,
+            },
+            randomness,
+            txs,
+        };
+
+        let prover = match MockProver::run(k, &circuit, power_of_randomness) {
```

Comment:
Just for curiosity, is or will be related to RLC in any way?

Reply:
Hm, I don't understand this question.  What would be related to RLC?  Could you clarify please?

Reply:
haha, I got confused with another thing. solved :)